### ISSUE TITLE
 how counting up a variable in range from 1 to n，isn't start 0

### ISSUE THREAD

**Issue by iwalkerr** [golang/go#31989]
```  
{{range $k,$v := .varList}}
      <tr>
           <td class='center' style="width: 30px;">{{$k}}</td>
     </tr>
{{end}}
```

**Comment by beoran** [comment]
The `text/template` package, for arrays of length n will iterate over indexes from 0 to (n-1) inclusive. This is the same as in the go language itself. Go language has 0 based arrays. See here for an example:

https://play.golang.org/p/2S-J45MP-xD

If iterating over .varList does not start with 0, perhaps it is not an array but a map?

**Comment by agnivade** [comment]
I think the OP is asking how to count a variable from 1 to n, and not 0 to n-1. 

@iwalkerr - You can use a helper function like this - https://play.golang.org/p/PX55SXLHUdm.

**Comment by iwalkerr** [comment]
Thanks




| |
walkerr
|
|
邮箱：walkerr@163.com
|
On 05/13/2019 16:26, Agniva De Sarker wrote:

Closed #31989.
